LEGO1/mxbitmap.cpp

```diff
@@ -0,0 +1,77 @@
+#include "mxbitmap.h"
+#include "mxbool.h"
+#include "mxpalette.h"
+#include "mxresult.h"
+
+// OFFSET: LEGO1 0x100bc980
+MxBitmap::MxBitmap()
+{
+  this->m_info = NULL;
+  this->m_bmiHeader = NULL;
+  this->m_paletteData = NULL;
+  this->m_data = NULL;
+  this->m_unk18 = MX_FALSE;
+  this->m_palette = NULL;
+}
+
+// OFFSET: LEGO1 0x100bca10
+MxBitmap::~MxBitmap()
+{
+  if (this->m_info != NULL) {
+    delete m_info;
+  }
+  if (this->m_data != NULL) {
+    delete m_data;
+  }
+  if (this->m_palette != NULL) {
+    delete m_palette;  
+  }
+}
+
+// OFFSET: LEGO1 0x100bd1c0
+MxPalette *MxBitmap::CreatePalette()
+{
+  // FIXME: This function needs MxPalette to be completed. Also INFERRING usage of MxBool
+  MxPalette *pal = NULL;
+  MxPalette *ppal;
+  MxBool success = MX_FALSE;
+
+  if(this->m_unk18 == MX_FALSE) {
+    // ppal = MxPalette::FromBitmapPalette(this->m_paletteData);
+  } else {
+    if(this->m_unk18 != MX_TRUE) {
+      if(!success && pal != NULL) {
+        delete pal;
+        pal = NULL;
+      }
+    }
+    //.pal = MxPalette::Clone(this->m_palette);
+  }
+  if(pal != NULL) {
+    success = MX_TRUE;
+  }
+
+  return pal;
+}
+
+// OFFSET: LEGO1 0x100bcd10
+long MxBitmap::Read(char const *filename)
+{
+  HANDLE handle;
+  int unk1;
+  MxResult ret = FAILURE;
+
+  handle = CreateFileA(filename,GENERIC_READ,FILE_SHARE_READ,(LPSECURITY_ATTRIBUTES)NULL,OPEN_EXISTING,FILE_ATTRIBUTE_NORMAL,(HANDLE)NULL);
+  if(handle != (HANDLE)INVALID_HANDLE_VALUE) {  // INVALID_HANDLE_VALUE = -1, or 0xffffffff
+		// FIXME: line 16. iVar gets changed in this line
+    if(unk1 == 0) {
+      ret = SUCCESS;
+    }
+  }
+  if(handle != (HANDLE)NULL) {
+    CloseHandle(handle);
+  }
+
```

Comment:
```suggestion
  handle = CreateFileA(filename, GENERIC_READ, FILE_SHARE_READ, (LPSECURITY_ATTRIBUTES)NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, (HANDLE)NULL);
  if (handle != (HANDLE)INVALID_HANDLE_VALUE && !this->VTable0x20()) {
    ret = SUCCESS;
  }
  if (handle != (HANDLE)NULL) {
    CloseHandle(handle);
  }

```

Reply:
The `if (handle != (HANDLE)NULL)` might deserve a comment to the effect of `// BUG compatibility with orig binary`.

Reply:
Sure, why is it a big?

Reply:
Zero is, however unbelievable (and in practice hopefully impossible to encounter), a valid HANDLE value. The only invalid is INVALID_HANDLE_VALUE.

Some nitpicking:
The comparison against `INVALID_HANDLE_VALUE` does not need a cast. The definition of `INVALID_HANDLE_VALUE` already includes the cast to `HANDLE`.

It's not very often one needs to be explicit about A/W version of API functions: `CreateFileA` -> `CreateFile`.

Reply:
The code right here is just temporary for now, we can come back to this later, this is an **Initial** implementation